### PR TITLE
Upgrade to Electron v32

### PR DESCRIPTION
**Pull request by sergiou87**
## Description

This update brings…
- The new API @yuzawa-san did for https://github.com/desktop/desktop/pull/17220 💖 
- Support for [CSS Anchor Positioning](https://developer.chrome.com/blog/new-in-chrome-125/#anchor-positioning) (as a native alternative to our Popover logic?)

It also updates node to v20.17 and had some type breaking change with our `ClickHandler` type.

I noticed it broke our `SandboxedMarkdown` and for some reason it was related to the behavior of `iframe` when `sandbox` is set to an empty string, or more specifically, when it does NOT have `allow-same-origin`. @niik and I discussed about it and think it should be safe 🤞 

Downgrading to v30 doesn't lose the list of repositories or any settings 🎉 

## Release notes

Notes: [Improved] Upgrade Electron to v32.1.2

### CODE REVIEW

**Comment by sergiou87** [comment]
I've noticed Markdown content (used in notifications, but also -on dev builds- in the PR previews) are not sized properly and therefore aren't visible.

**Review by tidy-dev — APPROVED** [review]
💖 I started doing a review last week then went down a rabbit hole trying to determine the implications of adding "allow-same-origin" to the sandbox. I vaguely remember discussing that it would be ok when we first implemented it, but we went without because at the time we could. It seems to me as long as we don't allow scripts, we are sandboxing from the main concern of users figuring out how to get script in the markdown past the sanitizers that accesses Desktop. 

I kicked it around macOS and it looks good!

**Comment by fisforfaheem** [comment]
Good! make sure to keep it updated to atleast 2 month older

File: script/validate-electron-version.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ type ChannelToValidate = 'production' | 'beta'
  */
 const ValidElectronVersions: Record<ChannelToValidate, string> = {
   production: '30.0.8',
-  beta: '30.0.8',
+  beta: '32.1.2',
```
Comment (Adi-alt19) [inline script/validate-electron-version.ts:22]: const: validelectrrones ,.

File: app/src/main-process/menu/build-default-menu.ts [hunk 2]
```diff
@@ -714,7 +714,10 @@ function emit(name: MenuEvent): ClickHandler {
     // while in DevTools. Since Desktop only supports one window at a time we
     // can be fairly certain that the first BrowserWindow we find is the one we
     // want.
-    const window = focusedWindow ?? BrowserWindow.getAllWindows()[0]
+    const window =
```
Comment (Adi-alt19) [inline app/src/main-process/menu/build-default-menu.ts:717]:
```suggestion

```

**Pull request merged**